dask_ml/model_selection/_incremental.py

```diff
@@ -198,34 +198,40 @@ async def _fit(
     else:
         y_test = await client.scatter(y_test)
 
-    # Convert to batches of delayed objects of numpy arrays
-    X_train = sorted(futures_of(X_train), key=lambda f: f.key)
-    y_train = sorted(futures_of(y_train), key=lambda f: f.key)
-    assert len(X_train) == len(y_train)
-
-    train_eg = await client.gather(client.map(len, y_train))
-    msg = "[CV%s] For training there are between %d and %d examples in each chunk"
-    logger.info(msg, prefix, min(train_eg), max(train_eg))
+    if hasattr(X_train, 'npartitions'):
```

Comment:
Can this if-statement be changed to `if isinstance(X_train, da.Array)`?

Reply:
the reason I don't want to restrict the if statement to be `isinstance(X_train, da.Array)` is because, potentially any  custom data structure like the hypotetical `CustomFrame` could have a `dask-like` API by implementing `npartitions` and `__dask_graph__`. 

In this case, for example, in the future I could further extend my `CustomFrame` (I or any other user) to work with data larger than memory and this would still work with `hyperband` even if is not a `da.Array`, because the API is compatible.

With this in mind maybe a good compromise could be:

`if dask.is_dask_collection(X_train)` ?

finally another reason why it might be better not to incorporate the if statement inside the `get_futures` function is because this will force you to check the condition every time you call the function which will not be ideal for `dask.collections` with a large number of partitions

Reply:
> if dask.is_dask_collection(X_train)

I think that's a better choice (though @TomAugspurger might have more input).

Reply:
Yeah, I think we `is_dask_collection` is better.